modules/metastorage/src/main/java/org/apache/ignite/metastorage/internal/MetaStorageManager.java

```diff
@@ -17,41 +17,62 @@
 
 package org.apache.ignite.metastorage.internal;
 
-import java.util.UUID;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import org.apache.ignite.configuration.internal.ConfigurationManager;
 import org.apache.ignite.configuration.schemas.runner.ClusterConfiguration;
 import org.apache.ignite.configuration.schemas.runner.LocalConfiguration;
 import org.apache.ignite.lang.IgniteUuid;
 import org.apache.ignite.metastorage.client.MetaStorageService;
 import org.apache.ignite.metastorage.common.Condition;
+import java.util.Collection;
+import org.apache.ignite.metastorage.common.Cursor;
 import org.apache.ignite.metastorage.common.Entry;
 import org.apache.ignite.metastorage.common.Key;
 import org.apache.ignite.metastorage.common.Operation;
 import org.apache.ignite.metastorage.common.WatchListener;
+import org.apache.ignite.metastorage.internal.watch.WatchAggregator;
 import org.apache.ignite.network.NetworkCluster;
 import org.apache.ignite.network.NetworkHandlersProvider;
 import org.apache.ignite.network.NetworkMessageHandler;
 import org.apache.ignite.raft.internal.Loza;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class MetaStorageManager {
+    /** */
     private final NetworkCluster network;
 
+    /** */
     private final Loza raftMgr;
 
+    /** */
     private final ConfigurationManager locConfigurationMgr;
 
-    private MetaStorageService service;
+    /** */
+    private final MetaStorageService svc;
 
+    /** */
+    private final WatchAggregator watchAggregator;
+
+    /** */
+    private CompletableFuture<IgniteUuid> igniteUuid;
+
+    /** */
+    private boolean deployed;
+
+    /** */
     public MetaStorageManager(
         NetworkCluster network,
         Loza raftMgr,
-        ConfigurationManager locConfigurationMgr
-    ) {
+        ConfigurationManager locConfigurationMgr,
+        MetaStorageService svc) {
```

Comment:
What's the point in propagation MetaStorageService through constructor? Why not to instantiate it internally?

Reply:
Well, until we have an IoC both approaches look acceptable to me.
However, this one seems better to me from a testing point of view. Mocking private final fields is tricky.